data/settings.js

```diff
@@ -2353,57 +2350,111 @@ the user's key pair.
 For EdDSA, you need to use X448 or X25519, case sensitive.`
 	},
 
-	mail_crypt_global_private_key: {
+	crypt_global_private_keys: {
+		plugin: 'mail-crypt',
+		values: setting_types.NAMED_LIST_FILTER,
+		seealso: [ 'crypt_private_key', 'crypt_private_key_password' ],
+		text: `
+List of private keys to decrypt files. Add [[setting,crypt_private_key]] and
+optionally [[setting,crypt_private_key_password]] inside each filter.`
+	},
+
+	crypt_global_public_key: {
 		plugin: 'mail-crypt',
 		values: setting_types.STRING,
 		text: `
-Private key(s) to decrypt files. Key(s) must be in PEM format, using pkey
-format.
+Public key to encrypt files. Key must be in
+[[link,mail_crypt_converting_ec_key_to_pkey,PEM pkey format]]. The PEM key may
+additionally be base64-encoded into a single line, which can make it easier to
+store into userdb extra fields.`
+	},
 
-You can define multiple keys by appending an increasing number to the
-setting label, e.g.: \`mail_crypt_global_private_key2\`.`
+	crypt_global_private_key: {
+		plugin: 'mail-crypt',
+		values: setting_types.NAMED_LIST_FILTER,
+		seealso: [ 'crypt_private_key', 'crypt_private_key_password' ],
+		text: `
+List of global private key(s) to decrypt mails. Add
+[[setting,crypt_private_key]] and optionally
+[[setting,crypt_private_key_password]] inside each filter.`
 	},
 
-	mail_crypt_global_public_key: {
+	crypt_user_key_encryption_key: {
 		plugin: 'mail-crypt',
-		values: setting_types.STRING,
+		values: setting_types.NAMED_LIST_FILTER,
+		seealso: [ 'crypt_private_key', 'crypt_private_key_password' ],
 		text: `
-Public key to encrypt files. Key must be in PEM format, using pkey format.`
+List of private key(s) to decrypt user's master private key. Add
+[[setting,crypt_private_key]] and optionally
+[[setting,crypt_private_key_password]] inside each filter.`
 	},
 
-	mail_crypt_private_key: {
+	crypt_user_key_password: {
 		plugin: 'mail-crypt',
 		values: setting_types.STRING,
 		text: `
-Private key to decrypt user's master key. Key must be in PEM format, using
-pkey format.`
+Password to decrypt user's master private key.`
 	},
 
-	mail_crypt_private_password: {
+	crypt_user_key_require_encrypted: {
+		plugin: 'mail-crypt',
+		values: setting_types.BOOLEAN,
+		seealso: [
+			'crypt_user_key_password',
+			'crypt_user_key_encryption_key',
+		],
+		text: `
+If yes, require user's master private key to be encrypted with
+[[setting,crypt_user_key_password]] or
+[[setting,crypt_user_key_encryption_key]]. If they are unset new user key
+generation will fail. This setting doesn't affect already existing
+non-encrypted keys.`
+	},
+
+	crypt_write_algorithm: {
 		plugin: 'mail-crypt',
 		values: setting_types.STRING,
+		default: 'aes-256-gcm-sha256',
 		text: `
-Password to decrypt user's master key or environment private key.`
+Set the encryption algorithm. If empty new mails are not encrypted, but
+existing mails can still be decrypted.`
```

Comment:
this setting would need better explanation somewhere about usable algorithm and mode combinations.

Reply:
I understand. I had this discussion with Timo earlier somewhere: I feel like improvements over these docs should be tracked in separate tickets. The focus of these PRs are to port the already made changes from the old to the new system. I explicitly don't want to do editorial changes.

Reply:
These could be added to `DOV-7009`